module/dkg/broker.go

```diff
@@ -206,16 +211,29 @@ func (b *Broker) Broadcast(data []byte) {
 // SubmitResult publishes the result of the DKG protocol to the smart contract.
 func (b *Broker) SubmitResult(groupKey crypto.PublicKey, pubKeys []crypto.PublicKey) error {
 
-	// If the DKG failed locally, we will get a nil key vector here. We need to convert
-	// the nil slice to a slice of nil keys before submission.
+	// If the DKG failed locally, we will get a nil key vector here.
+	// There are two different endpoints in the DKG smart contract for submitting a happy-path and failure-path result.
```

Comment:
```suggestion
	// There are two different transaction templates for submitting either a happy-path and failure-path result.
	// We use SubmitResult to submit a successful result and SubmitEmptyResult to communicate that we completed the DKG without a result.
```

Technically there is one "endpoint" (semi-public function accessible to DKG participants) and two transaction templates accepting different arguments for interacting with that function.